### DISCUSSION TITLE
docs/develop/getting-started/creating-an-addon/

### DISCUSSION THREAD

**Discussion by giscus** [ScratchAddons/website-v2#191] [Page comments]
# docs/develop/getting-started/creating-an-addon/

https://scratchaddons.com/docs/develop/getting-started/creating-an-addon/

**Comment by BarthdryZ** [comment]
Thanks for teaching about the json format.

**Comment by YahyaBotMaker** [comment]
Thanks for the tutorial, I am trying my best to make an addon!!!

**Reply by YahyaBotMaker** [reply]
I was wondering, is there a way to add blocks through the addons? I mean the code

**Reply by BroJac5246** [reply]
Yes, although I don't know how.

**Comment by SnappyRiffs** [comment]
Can i make an extra key support without breaking the already existing one?

**Reply by WorldLanguages** [reply]
You can do whatever you want, locally.
If you want your code to be added to the stable version of Scratch Addons, then we probably wouldn't accept 2 different addons that do the same thing.

**Reply by YahyaBotMaker** [reply]
can we add blocks through the addons?

**Reply by WorldLanguages** [reply]
@YahyaBotMaker You could check how the "debugger" addon adds its own blocks for inspiration.

**Reply by YahyaBotMaker** [reply]
ok thanks!

**Reply by Littlethief1** [reply]
i checked how the debug works and it's pretty to modify, but making a standalone addon is a whole other issue

**Comment by RussianRScratcher** [comment]
how do i can create a addon

**Reply by BroJac5246** [reply]
Follow these instructions. Once you're done, create a PR.

**Comment by PlayfulMathematician** [comment]
How do I do this in firefox

**Reply by BroJac5246** [reply]
I would highly recommend using Chrome for developing addons. Nevertheless, you can go to `about:debugging` on Firefox to load an extension from a folder.

**Reply by Samq64** [reply]
https://scratchaddons.com/docs/getting-started/installing/#installing-on-mozilla-firefox

**Comment by SnappyRiffs** [comment]
Scratch Addons crashed: invalid addon.json for addon with id <project id>. Click the "Errors" button on the extension tile for more details. How do I fix this?

**Reply by Samq64** [reply]
Check your `addon.json` file for syntax errors. It's usually an extra or missing comma.

**Comment by Luan364** [comment]
Thanks i hope i dont break a scratch rule i cinda forgot the guidlines
